vpr/src/route/rr_graph.cpp

```diff
@@ -2773,23 +2800,15 @@ static void add_pb_edges(RRGraphBuilder& rr_graph_builder,
                                               conn_pin_physical_num);
 
             if (is_remapped) {
-                bool found = false;
+                auto& all_sw_inf = g_vpr_ctx.mutable_device().all_sw_inf;
                 float delay = g_vpr_ctx.device().all_sw_inf.at(sw_idx).Tdel();
-                const auto& rr_switches = rr_graph_builder.rr_switch();
-                for (int sw_id = 0; sw_id < (int)rr_switches.size(); sw_id++) {
-                    const auto& rr_switch = rr_switches[RRSwitchId(sw_id)];
-                    if (rr_switch.intra_tile) {
-                        if (rr_switch.Tdel == delay) {
-                            sw_idx = sw_id;
-                            found = true;
-                            break;
-                        }
-                    }
-                }
-                // If the graph is loaded from a file, we expect that all sw types are already listed there since currently, we are not doing any further
-                // Optimization. If the optimization done when the rr graph file was generated is different from the current optimization, in the case that
-                // these optimizations create different RR switches, this VTR ASSERT can be removed.
-                VTR_ASSERT(found);
+                bool is_new_sw;
+                std::tie(is_new_sw, sw_idx) = find_create_intra_cluster_sw(rr_graph_builder,
```

Comment:
Check that this function has a doxygen comment and add one if it doesn't.

Reply:
Should also add or improve Doxygen comments for related functions like add_intra_tile_edges_rr_graph which don't have Doxygen comments, or have only placeholder ones that don't really describe the parameters etc.